optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java

```diff
@@ -151,7 +151,7 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         } else if (o instanceof SwapMove) {
-            SwapMove other = (SwapMove) o;
+            SwapMove<?> other = (SwapMove) o;
```

Comment:
We could do that but that would be an unchecked conversion which I wanted to avoid and the exact type of the move is not needed for the equality check.